### PR TITLE
Create analysis snapshot in dashboard

### PR DESCRIPTION
**Pull request by as2388**
I think this is the absolute minimum code needed to create usable snapshots

### CODE REVIEW

File: analysis_dashboard/analysis_dashboard.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        Client for accessing an Analysis Dashboard Firebase project.
+
+        :param firebase_app: Firebase app.
+        :type firebase_app: firebase_admin.App
+        """
+        self._firebase_app = firebase_app
+
+    @classmethod
+    def init_from_credentials(cls, cert, app_name="AnalysisDashboard"):
+        """
```
Comment (DanielMwendwa) [inline analysis_dashboard/analysis_dashboard.py:24]:
```suggestion
        """
        Initializes an AnalysisDashboard instance using Firestore service account certificate.
```

File: analysis_dashboard/analysis_dashboard.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        :param cert: Firestore service account certificate, as a path to a file or a dictionary.
+        :type cert: str | dict
+        :param app_name: Name to give the Firebase app instance used to connect.
+        :type app_name: str
+        :return:
```
Comment (DanielMwendwa) [inline analysis_dashboard/analysis_dashboard.py:29]:
```suggestion
        :return: Initialized AnalysisDashboard instance.
```

File: analysis_dashboard/analysis_dashboard.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        :rtype: AnalysisDashboard
+        """
+        return cls(initialize_firebase_app(cert, app_name))
```
Comment (DanielMwendwa) [inline analysis_dashboard/analysis_dashboard.py:32]: `firebase_app = initialize_firebase_app(cert, app_name)
return cls(firebase_app)`

**Review by DanielMwendwa — APPROVED** [review]

File: analysis_dashboard/analysis_dashboard.py (hunk 1 above)
Comment (as2388) [reply analysis_dashboard/analysis_dashboard.py:24]: Thanks, I've updated this docstring

File: analysis_dashboard/analysis_dashboard.py (hunk 2 above)
Comment (as2388) [reply analysis_dashboard/analysis_dashboard.py:29]: updated

File: analysis_dashboard/analysis_dashboard.py (hunk 3 above)
Comment (as2388) [reply analysis_dashboard/analysis_dashboard.py:32]: changed

File: analysis_dashboard/data_models/analysis_snapshot.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import uuid
+
+
+class AnalysisSnapshot:
+    def __init__(self, files, snapshot_id=None):
+        """
+        Represents a single version of a piece of analysis, describing which datasets are available and how those
+        datasets were generated.
```
[the remark this answers is not in the document]
Comment (as2388) [reply analysis_dashboard/data_models/analysis_snapshot.py:8]: I don't think we need to explain what an __init__ method does

File: analysis_dashboard/data_models/analysis_snapshot.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import uuid
+
+
+class AnalysisSnapshot:
```
[the remark this answers is not in the document]
Comment (as2388) [reply analysis_dashboard/data_models/analysis_snapshot.py:4]: We have always put the class docstring in the init method, but I think you are right that the description should go here. I'm happy for us to have an overview docstring on that class itself from now on, and will change this file.

File: analysis_dashboard/data_models/analysis_snapshot.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+                           the constructor.
+        :type snapshot_id: str | None
+        """
+        if snapshot_id is None:
+            snapshot_id = str(uuid.uuid4())
+
+        self.snapshot_id = snapshot_id
+        self.datasets = files
+
+    def to_dict(self):
```
[the remark this answers is not in the document]
Comment (as2388) [reply analysis_dashboard/data_models/analysis_snapshot.py:25]: updated

**Pull request merged**